src/auth/SignInComponent.js

```diff
@@ -1,91 +1,109 @@
+import React, { useContext } from "react";
+import { Button, TextField, Link, Grid, Box, Typography } from "@mui/material";
+import { AuthContext } from "./AuthContext";
+import { useNavigate } from "react-router-dom";
 
 function SignInComponent() {
-
   const navigate = useNavigate();
+  const { isLoggedIn, login, logout } = useContext(AuthContext);
 
   const handleSubmit = (event) => {
     event.preventDefault();
     const data = new FormData(event.currentTarget);
     const loginData = {
-      username: data.get('email'),
-      password: data.get('password'),
+      username: data.get("email"),
+      password: data.get("password"),
     };
 
     fetch("http://localhost:24001/api/axion-auth-service/public/auth/login", {
       method: "post",
-      headers: {
-        "Content-Type": "application/json"
-      },
+      headers: { "Content-Type": "application/json" },
       credentials: "include",
-      body: JSON.stringify(loginData)
+      body: JSON.stringify(loginData),
     })
       .then((response) => {
         if (!response.ok) {
-          throw new Error('HTTP error!'); // TODO handle error gracefully
+          throw new Error("HTTP error!"); // TODO handle error gracefully
         }
         return response.text();
       })
-      .then(() => navigate("/"));
-
+      .then((token) => {
+        console.log("User logged in, setting login to true");
+        login(token);
+        navigate("/");
+      })
+      .catch((err) => console.error("Login faild:", err));
   };
 
   return (
     <Box
       sx={{
         marginTop: 8,
-        display: 'flex',
-        flexDirection: 'column',
-        alignItems: 'center',
+        display: "flex",
+        flexDirection: "column",
+        alignItems: "center",
       }}
     >
-      <Typography component="h1" variant="h5">
-        Sign in
-      </Typography>
-      <Box component="form" onSubmit={handleSubmit} noValidate sx={{mt: 1}}>
-        <TextField
-          margin="normal"
-          required
-          fullWidth
-          id="email"
-          label="Email Address"
-          name="email"
-          autoComplete="email"
-          autoFocus
-        />
-        <TextField
-          margin="normal"
-          required
-          fullWidth
-          name="password"
-          label="Password"
-          type="password"
-          id="password"
-          autoComplete="current-password"
-        />
-        <Button
-          type="submit"
-          fullWidth
-          variant="contained"
-          sx={{mt: 3, mb: 2}}
-        >
-          Sign In
-        </Button>
-        <Grid container>
-          <Grid item xs>
-            <Link href="#" variant="body2">
-              Forgot password?
-            </Link>
-          </Grid>
-          <Grid item>
-            <Link href="#" variant="body2">
-              {"Don't have an account? Sign Up"}
-            </Link>
-          </Grid>
-        </Grid>
-      </Box>
+      {!isLoggedIn ? (
```

Comment:
This part should be reverted. We should not check `isLoggedIn` here. It was moved to Header component.

Reply:
Fixed